### DISCUSSION TITLE
Numerous CVE in apache/airflow Docker image

### DISCUSSION THREAD

**Discussion by sipr-invivo** [apache/airflow#40649] [General]
### Apache Airflow version

2.9.2

### What happened?

There are numerous vulnerabilities in the Docker image apache/airflow, even in the slim version of the image, as reported by Aquasec Trivy

```
docker.io/apache/airflow:slim-2.9.2 (debian 12.5)

Total: 19 (HIGH: 19, CRITICAL: 0)
```

Note that there are also 4 vulnerabilities in Python packages

```
Python (python-pkg)

Total: 4 (HIGH: 4, CRITICAL: 0)
```

### What you think should happen instead?

No vulnerabilities

### How to reproduce

To restrict vulnerabilities only to fixed ones, HIGH & CRITICAL, and on the os side

```
trivy image --scanners vuln --ignore-unfixed --severity HIGH,CRITICAL --vuln-type os,library docker.io/apache/airflow:slim-2.9.2
```

### Operating System

N/A

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
See https://airflow.apache.org/docs/docker-stack/index.html#fixing-images-at-release-time - this explains what is our approach to releasing new images and How YOU can build newer images of older airlfow versions if you do not want to wait for newer versions if airlfow.

It contains full explanation how it works and how you as a commercial user of open-source software can help in making airlfow secure - by contributing your and your company engineering time to fix the issues if there are  and provide PR (but also explains how you can manage latest versions in your own images that you can build for yourself).

If you have reasons to believe that latest main version of airflow has security issues that should be handled and are not, feel free to follow the security policy of Airlfow https://github.com/apache/airflow/security/policy and report such issues ONE PER ONE EMAIL - including the POC on how each security issue affects airflow (see the link above for explanation that relying on scanners and assuming that software is vullnerable simply because image contains a dependency is a bad idea). 

Opening such issues serves nothing but noise, unless you are willing to spend time and anergy on analysing your scans and provide specific individualk reports for each issue such scanner find - which is probably the least you and your company shoudl do in return for the free software you get.

Thank you for your future cooperation and contributions to Airflow's security (assuming that you follow the security policy).
